presto-main/src/main/java/com/facebook/presto/operator/PageBufferClient.java

```diff
@@ -0,0 +1,552 @@
+                catch (PrestoException e) {
+                    handleFailure(e, resultFuture);
+                    return;
+                }
+
+                synchronized (PageBufferClient.this) {
+                    // client is complete, acknowledge it by sending it a delete in the next request
+                    if (result.isClientComplete()) {
+                        completed = true;
+                    }
+                    if (future == resultFuture) {
+                        future = null;
+                    }
+                    lastUpdate = DateTime.now();
+                }
+                requestsCompleted.incrementAndGet();
+                clientCallback.requestComplete(PageBufferClient.this);
+            }
+
+            @Override
+            public void onFailure(Throwable t)
+            {
+                log.debug("Request to %s failed %s", uri, t);
+                checkNotHoldsLock(this);
+
+                t = resultClient.rewriteException(t);
+                if (!(t instanceof PrestoException) && backoff.failure()) {
+                    String message = format("%s (%s - %s failures, failure duration %s, total failed request time %s)",
+                            WORKER_NODE_ERROR,
+                            uri,
+                            backoff.getFailureCount(),
+                            backoff.getFailureDuration().convertTo(SECONDS),
+                            backoff.getFailureRequestTimeTotal().convertTo(SECONDS));
+                    t = new PageTransportTimeoutException(fromUri(uri), message, t);
+                }
+                handleFailure(t, resultFuture);
+            }
+        }, pageBufferClientCallbackExecutor);
+    }
+
+    private synchronized void sendDelete()
+    {
+        ListenableFuture<?> resultFuture = resultClient.abortResults();
+        future = resultFuture;
+        Futures.addCallback(resultFuture, new FutureCallback<Object>()
+        {
+            @Override
+            public void onSuccess(@Nullable Object result)
+            {
+                checkNotHoldsLock(this);
+                backoff.success();
+                synchronized (PageBufferClient.this) {
+                    closed = true;
+                    if (future == resultFuture) {
+                        future = null;
+                    }
+                    lastUpdate = DateTime.now();
+                }
+                requestsCompleted.incrementAndGet();
+                clientCallback.clientFinished(PageBufferClient.this);
+            }
+
+            @Override
+            public void onFailure(Throwable t)
+            {
+                checkNotHoldsLock(this);
+
+                log.error("Request to delete %s failed %s", location, t);
+                if (!(t instanceof PrestoException) && backoff.failure()) {
+                    String message = format("Error closing remote buffer (%s - %s failures, failure duration %s, total failed request time %s)",
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format long: &#39;getFailureCount&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 405. Consider formatting this data with &#39;%d&#39; instead.